corehq/motech/generic_inbound/views.py

```diff
@@ -215,3 +225,37 @@ def _execute_case_api(domain, couch_user, device_id, context, api_model):
         'form_id': xform.form_id,
         'case': serialize_case(case_or_cases),
     }
+
+
+def _log_api_request(api, request, response):
+    is_success = response.status_code == 200
+    status = RequestLog.Status.SUCCESS if is_success else RequestLog.Status.ERROR
+    response_body = response.content.decode('utf-8')
+    log = RequestLog.objects.create(
+        domain=request.domain,
+        api=api,
+        status=status,
+        response_status=response.status_code,
+        error_message=response_body if not is_success else '',
+        username=request.couch_user.username,
+        request_method=request.method,
+        request_query=request.get_raw_uri(),
+        request_body=request.body.decode('utf-8'),
+        request_headers=dict(request.headers),
+        request_ip=get_ip(request),
+    )
+
+    response_json = json.loads(response.content)
+    if is_success:
+        case_ids = [c['case_id'] for c in
+                    response_json.get('cases', [response_json.get('case')])]
```

Comment:
this is what I'm talking about in my PR message - I don't love parsing this sort of info out from a response object, but will defer perfecting that until the "retry" PR.

Reply:
A simpler option would be to make `_generic_inbound_api` return custom response object that includes the data you need